<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">@import url('https://themes.googleusercontent.com/fonts/css?kit=tY3cJZ8eDORKFMyGha79redu8NCjfCLmCp0sVMGUDOs');ol{margin:0;padding:0}table td,table th{padding:0}.c30{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:13pt;font-family:"Arial";font-style:normal}.c31{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:14pt;font-family:"Arial";font-style:normal}.c1{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c2{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left;height:11pt}.c7{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c6{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c21{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:18pt;font-family:"Arial";font-style:normal}.c4{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:italic}.c24{background-color:#ffffff;-webkit-text-decoration-skip:none;color:#1155cc;font-weight:400;text-decoration:underline;text-decoration-skip-ink:none;font-family:"Roboto"}.c3{padding-top:0pt;padding-bottom:12pt;line-height:1.4166666666666667;orphans:2;widows:2;text-align:left}.c27{padding-top:0pt;padding-bottom:0pt;line-height:1.1500022727272727;orphans:2;widows:2;text-align:left}.c34{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial"}.c8{padding-top:0pt;padding-bottom:12pt;line-height:1.8333333333333333;orphans:2;widows:2;text-align:left}.c32{padding-top:0pt;padding-bottom:10pt;line-height:1.1500022727272727;orphans:2;widows:2;text-align:left}.c5{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c25{color:#500050;text-decoration:none;vertical-align:baseline;font-family:"Arial";font-style:normal}.c35{font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Roboto";font-style:normal}.c16{font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Arial";font-style:normal}.c26{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#0000ff;text-decoration:underline}.c14{background-color:#ffffff;max-width:468pt;padding:72pt 72pt 72pt 72pt}.c18{font-size:7pt;font-family:"Times New Roman";font-weight:400}.c29{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;text-decoration:underline}.c0{vertical-align:super;font-size:12pt;color:#404040}.c13{background-color:#ffffff;font-style:italic;color:#212121}.c36{background-color:#ffffff;color:#212121}.c17{background-color:#ffffff;color:#2c3e50}.c23{color:inherit;text-decoration:inherit}.c10{color:#404040;font-size:12pt}.c20{font-family:"Cambria";font-weight:400}.c9{font-size:12pt;font-weight:700}.c28{background-color:#ffffff;color:#222222}.c15{margin-left:36pt}.c33{color:#0000ff}.c22{color:#ff0000}.c11{height:11pt}.c12{font-size:12pt}.c19{font-style:italic}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c14 doc-content"><p class="c8"><span class="c31">Genomics of Human Populations (NYU, Fall 2022)</span></p><p class="c8"><span class="c31">Syllabus and Schedule</span></p><p class="c3"><span class="c9">INSTRUCTOR: </span><span class="c1">Professor Jonathan M. Flowers, Ph.D. </span></p><p class="c3"><span class="c9">OFFICE:</span><span class="c1">&nbsp;Center for Genomics and Systems Biology, 12 Waverly Place, 6th Floor, Room 607</span></p><p class="c3"><span class="c9">EMAIL:</span><span class="c12">&nbsp; </span><span class="c16 c12 c33">jmf11@nyu.edu</span></p><p class="c3"><span class="c9">SESSIONS AND LOCATION: </span><span class="c12">Mondays and Wednesdays 12:30-1:45 Bldg: </span><span class="c12 c17">GCASL, Room 275</span><span class="c1">, Washington Square</span></p><p class="c3"><span class="c9">OFFICE HOURS:</span><span class="c1">&nbsp;2 scheduled hours per week or by appointment.</span></p><p class="c3"><span class="c6">COURSE OVERVIEW:</span></p><p class="c3"><span class="c1">This course covers topics in the field of human population genetics including human ancestry and admixture, human demography, linkage disequilibrium, genome-wide association studies (GWAS), genetic architecture of human traits and diseases, natural selection in the human genome, and application of population genomics to the study of cancer and disease. The course includes lecture and recitation components with the latter geared to teaching students basic skills in population genomic data analysis.</span></p><p class="c3"><span class="c1">By the end of the course students will be able to:</span></p><p class="c3 c15"><span class="c20 c12">&middot;</span><span class="c18">&nbsp;</span><span class="c1">Understand principles of population genetics theory and its application to human population genomic datasets</span></p><p class="c3 c15"><span class="c20 c12">&middot;</span><span class="c18">&nbsp; </span><span class="c1">Conduct analysis of human population genomic datasets using standard methodological approaches</span></p><p class="c3 c15"><span class="c12 c20">&middot;</span><span class="c18">&nbsp; </span><span class="c1">Draw biological inferences from human population genomic data</span></p><p class="c3"><span class="c6">PREREQUISITES:</span></p><p class="c5"><span class="c1">Undergraduate Molecular Biology and Genetics; Introductory Statistics with R is recommended</span></p><p class="c5"><span class="c1">&nbsp;</span></p><p class="c5"><span class="c1">&nbsp;</span></p><p class="c3"><span class="c6">COURSE DETAILS: </span></p><p class="c3"><span class="c12">The course consists of two sessions per week. </span><span class="c1">Each week will be roughly divided into i) a lecture with discussion of assigned readings and ii) a recitation/workshop. &nbsp;The assigned reading will draw from a variety of sources including textbooks and primary research papers. &nbsp;The recitation/workshop section will consist of a hands-on tutorial on how to conduct population genomic data analysis using either the R statistical programming language or using third party software tools.</span></p><p class="c3"><span class="c1">Introductory level coding experience in R is recommended for the course. A laptop computer will be needed. &nbsp;If you do not have access to one let the instructor know so arrangements can be made.</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">COMPUTER AND COMPUTATION ENVIRONMENT:</span></p><p class="c3"><span class="c1">Each student will need regular access to a personal computer. Computation for the course will be conducted primarily using RStudio and the R Statistical Programming Language. Each student will need to install RStudio on their computer for this purpose. Additional details and orientation will be provided at the first recitation. Please inform your instructor if there are any issues with this requirement.</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">TEXT:</span></p><p class="c3"><span class="c1">Molecular Population Genetics (2019) by Matthew W. Hahn, Oxford University Press</span></p><p class="c32"><span class="c10">The book, </span><span class="c10 c29">Molecular Population Genetics</span><span class="c10">&nbsp;by Hahn, will be delivered to you digitally. The cost of the book is $40.75, which will be added as a &ldquo;book charge&rdquo; to your bursar bill, this is a savings of $</span><span class="c12">79.25 over</span><span class="c10">&nbsp;the publisher&rsquo;s hardcopy list price. This charge will post to your bursar account on August 18</span><span class="c0">th </span><span class="c10 c16">or after you have registered to the course.</span></p><p class="c27 c11"><span class="c7"></span></p><p class="c27"><span class="c10">Should you choose to remove yourself from the program and find your course materials elsewhere, you must login </span><span class="c12 c26"><a class="c23" href="https://www.google.com/url?q=https://includedcp.follett.com/2015&amp;sa=D&amp;source=editors&amp;ust=1662669286432940&amp;usg=AOvVaw0aunQ_-jNI_qwPDVBwb3kz">here to the student portal</a></span><span class="c16 c10">&nbsp;and opt out of having the course materials provided to you by September 16th. If you opt out or drop the class your account will be refunded in 24-48 hours.</span></p><p class="c27 c11"><span class="c16 c10"></span></p><p class="c27"><span class="c16 c10">Hard copies of the text can also be obtained at the bookstore if desired.</span></p><p class="c3 c11"><span class="c9 c25"></span></p><p class="c3"><span class="c6">GRADING:</span></p><p class="c3"><span class="c1">Quizzes&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10 points each (x 8) = 80 points - (2 lowest scores)* = 60 points &nbsp;</span></p><p class="c3"><span class="c1">Homework &nbsp; &nbsp; 30 points each (x 4) = 120 points</span></p><p class="c3"><span class="c1">Participation&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10 points (x 2)** = 20 points</span></p><p class="c3"><span class="c1">Total = 200 points</span></p><p class="c3"><span class="c1">*the lowest 2 quiz scores will be dropped</span></p><p class="c3"><span class="c1">**evaluated separately for Weeks 1-7 and 8-14.</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">ASSESSMENTS:</span></p><p class="c3"><span class="c1">Assessments for the course include 8 quizzes which will assess student&rsquo;s understanding of key concepts from the lectures. The 4 homework assignments provide students with an opportunity to analyze real population genomic dataset and draw biological inferences from the data. These homeworks will allow students to apply methods introduced in the recitation sessions and conduct analysis and interpretation using concepts introduced in the lectures.</span></p><p class="c3"><span class="c1">Four problem sets will be assigned</span></p><p class="c3"><span class="c1">Eight Quizzes will serve as knowledge checks from readings and lectures from the previous week(s) material.</span></p><p class="c3"><span class="c1">&ldquo;Participation&rdquo; is evaluated as follows:</span></p><p class="c3 c15"><span class="c20 c12">&middot;</span><span class="c18">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c1">Preparation of questions/criticisms/ideas based on the assigned reading.</span></p><p class="c3 c15"><span class="c20 c12">&middot;</span><span class="c18">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c1">Responding to other students&rsquo; questions/criticisms/ideas.</span></p><p class="c3 c15"><span class="c20 c12">&middot;</span><span class="c18">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c1">Participation in discussions in the recitation</span></p><p class="c3 c15"><span class="c1">Students will be evaluated separately for first half and second half of course (10 points each half)</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">COVID-19 POLICY:</span></p><p class="c3"><span class="c1">Student safety will continue to be the top priority at NYU as it is for students attending this course. Fall 2022 classes are scheduled to be held in person. Students who must take all, or part, of the course remotely due to Covid-19 related issues, will be accommodated. However, students will still be required to attend the two weekly sessions live (synchronously) and complete quizzes in real time online at the same time as students attending in person. If you have any special Covid-related circumstances, please contact your instructor and we will do what we can to accommodate you.</span></p><p class="c3 c11"><span class="c1"></span></p><p class="c3"><span class="c6">UNIVERSITY POLICIES:</span></p><p class="c3"><span class="c12">Students are expected to know and understand the policies on academic integrity, including University, CAS, and GSAS policies: </span><span class="c26 c12">https://www.nyu.edu/about/policies-guidelines-compliance/policies-and-guidelines/academic-integrity-for-students-at-nyu.html</span><span class="c12">&nbsp;; </span><span class="c26 c12">https://cas.nyu.edu/content/nyu-as/cas/academic-integrity.html; </span><span class="c12">Also see the GSAS Statement on Academic Integrity: </span><span class="c26 c12">http://gsas.nyu.edu/content/nyu-as/gsas/about-gsas/policies-and-procedures/gsas-statement-on-academic-integrity.html</span><span class="c1">.</span></p><p class="c3 c11"><span class="c1"></span></p><p class="c3"><span class="c1">If a student is caught cheating or plagiarizing, the Instructor may, at her or his discretion, give the student an academic sanction, which may include a warning and/or reduction of the grade on an assessment item (e.g. exam) or even the final course grade (in consultation with the Director of the Master&rsquo;s Program, who may meet with the faculty and the student to discuss the nature of the offense). Depending on the severity of the infraction, this could even mean failure of the student in the course. The student may appeal any grade reduction to the Director of the Master&rsquo;s Program. The departmental decision is final. In addition, any substantial case brought to the Director of the Master&rsquo;s Program must be referred to the Dean&#39;s office for possible disciplinary action.</span></p><p class="c3"><span class="c1">If you have any questions or uncertainties about these policies, please consult the Instructor, Director of the Master&rsquo;s Program, or Dean&rsquo;s office.</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">COURSE SCHEDULE</span></p><p class="c3"><span class="c9">WEEK 1: Introduction to Population Genomics I</span></p><p class="c5"><span class="c6">No Class Monday 9/05/2022 12:30 - 1:45 pm (Labor Day)</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c1"></span></p><p class="c3"><span class="c6">Wednesday 9/07/2022 12:30 - 1:45 pm</span></p><p class="c3"><span class="c1">Lecture: Introduction to population genomics</span></p><p class="c3"><span class="c1">Recitation: No recitation this week</span></p><p class="c3"><span class="c1">Reading: Hahn Chapters 1 and 2</span></p><p class="c2"><span class="c7"></span></p><p class="c3"><span class="c6">WEEK 2: Introduction to Population Genomics II</span></p><p class="c3"><span class="c6">Monday 9/12/2022 12:30 - 1:45 pm</span></p><p class="c5"><span class="c1">Lecture: Molecular population genetics: Description of DNA sequence variation using population summary statistics</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Hahn Chapter 3 &ldquo;Describing Variation&rdquo;</span></p><p class="c5"><span class="c1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; &nbsp;</span></p><p class="c5"><span class="c6">Wednesday 9/14/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Recitation: Population genomic data processing in R</span></p><p class="c2"><span class="c1"></span></p><p class="c3"><span class="c12">Homework: Problem set 1 (*due end of Week 4)</span></p><p class="c5"><span class="c9 c19">Quiz 1: Weeks 1 and 2 lecture materials</span></p><p class="c2"><span class="c7"></span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 3: Recombination and linkage disequilibrium in the human genome</span></p><p class="c5"><span class="c6">Monday 9/19/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Linkage disequilibrium: concepts and inference</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Hahn Chapter 4 &ldquo;Recombination&rdquo;</span></p><p class="c2"><span class="c7"></span></p><p class="c2"><span class="c4"></span></p><p class="c5"><span class="c6">Wednesday 9/21/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Recitation: Calculation of population summary statistics and linkage disequilibrium &nbsp;in R</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Homework: Problem set 1 (*due Week 4)</span></p><p class="c2"><span class="c1"></span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 4: Population Structure in Human Populations I</span></p><p class="c5"><span class="c6">Monday 9/26/2022, 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture Wahlund Effect and population subdivision</span></p><p class="c2"><span class="c1"></span></p><p class="c3"><span class="c1">Reading: Hahn Chapter 5 &ldquo;Population Structure&rdquo; (pp 79-104)</span></p><p class="c5"><span class="c4">Quiz 2: Week 3 lecture materials</span></p><p class="c2"><span class="c6"></span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Wednesday 9/28/2022, 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c3"><span class="c1">Recitation: Calculating summaries of population differentiation in R</span></p><p class="c3"><span class="c4">Homework: Problem set 1 due</span></p><p class="c3 c11"><span class="c1"></span></p><p class="c3"><span class="c6">WEEK 5: Population Structure in Human Populations II</span></p><p class="c5"><span class="c6">Monday 10/3/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Population genomic approaches to identifying populations</span></p><p class="c2"><span class="c7"></span></p><p class="c3"><span class="c1">Reading: Hahn Chapter 5 &ldquo;Population Structure&rdquo; (pp. 104-110)</span></p><p class="c3"><span class="c1">Lawson et al. 2018. A tutorial on how not to over-interpret STRUCTURE and ADMIXTURE barplots. Nature Communications. 9:3258</span></p><p class="c3"><span class="c1">Novembre et al. 2008. Genes mirror geography within Europe. Nature. 456: 98-101</span></p><p class="c3"><span class="c1">Homework: Problem set 2 (*due Week 7)</span></p><p class="c5"><span class="c6">Wednesday 10/5/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c3"><span class="c1">Recitation Population structure inferred from model-based clustering and PCA </span></p><p class="c3"><span class="c1">Homework: Problem Set 2 (*due Week 7)</span></p><p class="c3"><span class="c4">Quiz 3: Weeks 4 and 5 lecture materials</span></p><p class="c3 c11"><span class="c1"></span></p><p class="c3"><span class="c6">WEEK 6: Inbreeding in human populations</span></p><p class="c5"><span class="c6">No Class Monday 10/10/2022 12:30 - 1:45 pm (Fall Break)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c9">Tuesday 10/11/2022 12:30 - 1:45 pm</span><span class="c12">&nbsp; </span><span class="c6">*Monday course schedule*</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Lecture: Inbreeding in human populations</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Hartl and Clark, pp. 135-147</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c28 c12">Ceballos, F., Joshi, P., Clark, D. </span><span class="c12 c19 c28">et al.</span><span class="c28 c12">&nbsp;Runs of homozygosity: windows into population history and trait architecture. </span><span class="c28 c12 c19">Nat Rev Genet</span><span class="c28 c12">&nbsp;19, 220&ndash;234 (2018).</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Wednesday 10/12/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c12">Recitation: TBA</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Homework: Problem set 2 (*due Week 7)</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c1"></span></p><p class="c3"><span class="c6">WEEK 7: Admixture Between Human Populations</span></p><p class="c5"><span class="c6">Monday 10/17/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c16 c12 c22"></span></p><p class="c5"><span class="c1">Lecture: Methods to infer admixture using population genomic data</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Hahn Chapter 9 (p. 222-248)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Green et al. 2010. A Draft Sequence of the Neanderthal Genome. Science. 328: 710-722</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c12">Reich D et al. 2009. Reconstructing Indian Population History. Nature. 461:489-494</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c4">Homework: Problem set 2 due</span></p><p class="c2"><span class="c6"></span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Wednesday 10/19/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Recitation: Application of admixture statistics and the inference of introgression using ADMIXTOOLS 2 R package</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Homework: Problem set 2 due</span></p><p class="c2"><span class="c4"></span></p><p class="c5"><span class="c4">Quiz 4: Week 6 and 7 lecture materials</span></p><p class="c2"><span class="c1"></span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 8: Genomic Studies of Human Population Demography I</span></p><p class="c3"><span class="c6">Monday 10/24/2022 12:30 - 1:45 pm</span></p><p class="c3"><span class="c1">Lecture: Human population demography and its inference I</span></p><p class="c5"><span class="c1">Readings: Hahn Chapter 6 and 9 (pp. 203-222)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Henn et al. 2012. The great human expansion. PNAS. 44:17758-17764.</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c6">Wednesday 10/26/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Recitation: TBA</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Homework: Problem set 3 (*due Week 12)</span></p><p class="c2"><span class="c1"></span></p><p class="c3 c11"><span class="c1"></span></p><p class="c3"><span class="c9">WEEK 9: Genomic Studies of Human Population Demography II</span></p><p class="c5"><span class="c9">Monday 10/31/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Human population demography and its inference II</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: TBA</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c9">Wednesday 11/2/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Recitation: TBA</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Homework: Problem set 3 (*due Week 12)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Quiz 5: Week 8 and 9 lecture Materials</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 10: Adaptation in the Human Genome I</span></p><p class="c5"><span class="c9">Monday 11/7/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Inferring direct selection using population genomic methods</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Hahn Chapters 7 &ldquo;Direct Selection&rdquo;</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Wednesday 11/9/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Recitation: Methods in selection inference I</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Homework: Problem set 3 (*due Week 12)</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c7"></span></p><p class="c3"><span class="c6">WEEK 11: Adaptation in the Human Genome II</span></p><p class="c5"><span class="c6">Monday 11/14/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Lecture: Inferring linked selection using population genomic methods</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Reading: Hahn Chapters 8 &ldquo;Linked Selection&rdquo;</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c6">Wednesday 11/16/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Recitation: Methods in selection inference II</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Homework: Problem set 3 (*due Week 12)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Quiz 6: Week 10 and 11 lecture Materials</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c16 c22 c12"></span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 12: &nbsp;Population Genomics of Disease I</span></p><p class="c5"><span class="c6">Monday 11/21/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Genetic load and deleterious burden in human populations I</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Reading: Henn B et al. 2015: Estimating the mutation load in human genomes. Nature Reviews Genetics. 16:33-43.</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Homework: Problem Set 3 due</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c6">Wednesday 11/23/2022 12:30 - 1:45 pm </span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c9">*no class, Fall Break*</span></p><p class="c2"><span class="c7"></span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">WEEK 13: Population Genomics of Disease II</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Monday 11/28/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Lecture: Genetic load and deleterious burden in human populations II</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c12">Reading: Readings: Pereira L et al. 2021. African genetic diversity and adaptation inform a precision medicine agenda. Nature Reviews Genetics. </span><span class="c24 c12"><a class="c23" href="https://www.google.com/url?q=https://doi.org/10.1038/s41576-020-00306-8&amp;sa=D&amp;source=editors&amp;ust=1662669286446466&amp;usg=AOvVaw211NHTU2D41JYmOVZcf_rS">https://doi.org/10.1038/s41576-020-00306-8</a></span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c12">Benton ML. et al. 2021. The influence of evolutionary history on human health and disease. Nature Reviews Genetics. </span><span class="c12 c24"><a class="c23" href="https://www.google.com/url?q=https://doi.org/10.1038/s41576-020-00305-9&amp;sa=D&amp;source=editors&amp;ust=1662669286446867&amp;usg=AOvVaw0jsOiWdfy4ldfRrRvXqxWf">https://doi.org/10.1038/s41576-020-00305-9</a></span></p><p class="c2"><span class="c35 c28 c12"></span></p><p class="c5"><span class="c1">Gurdasani D. et al. 2019. Genomics of disease risk in globally diverse populations. Nature Reviews Genetics. 20:520-535.</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c6">Wednesday 11/30/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Recitation: TBA</span></p><p class="c2"><span class="c7"></span></p><p class="c5"><span class="c1">Homework: Problem set 4 (*due Week 14)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Quiz 7: Week 12 and 13 Lecture Materials</span></p><p class="c2"><span class="c1"></span></p><p class="c2"><span class="c7"></span></p><p class="c3"><span class="c6">WEEK 14: GWAS I: Trait Mapping with Case-control Studies</span></p><p class="c5"><span class="c6">Monday 12/5/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c12">Lecture:</span><span class="c9">&nbsp;</span><span class="c1">Experimental design of case control studies, statistical analysis of case control data</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c12">Reading: Pearson TA, Manolio TA. 2008. How to interpret a genomewide association study. JAMA. 299:1335-1344.</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c6">Wednesday 12/7/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c12">Recitation: Genome-wide association analysis II with snpStats R package I</span></p><p class="c2"><span class="c21"></span></p><p class="c5"><span class="c12">Reading: </span><span class="c12 c36">Reed, E., Nunez, S., Kulp, D., Qian, J., Reilly, M. P., &amp; Foulkes, A. S. (2015). A guide to genome-wide association analysis and post-analytic interrogation. </span><span class="c13 c12">Statistics in medicine</span><span class="c36 c12">, </span><span class="c12 c13">34</span><span class="c36 c12">(28), 3769&ndash;3792.</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Homework: Problem set 4 (due: TBA)</span></p><p class="c3 c11"><span class="c6"></span></p><p class="c3"><span class="c6">WEEK 15: GWAS II: Trait Mapping with Population-based Designs</span></p><p class="c3"><span class="c6">Monday 12/12/2022 12:30 - 1:45 pm</span></p><p class="c5"><span class="c1">Lecture: Approaches to controlling for population structure and kinship with mixed linear models and the application of GWAS in human biology</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Visscher PM et al. 2017. 10 years of GWAS discovery: Biology, function, translation. Am. J. Human Genet. 101:5-22.</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c1">Bycroft C. et al. 2018. The UK Biobank resource with deep phenotyping and genomic data. Nature. 562:203-209.</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c6">Wednesday 12/14/2022 12:30 - 1:45 pm</span></p><p class="c2"><span class="c6"></span></p><p class="c5"><span class="c1">Recitation: Genome-wide association analysis II with snpStats R package I</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c4">Homework: Problem set 4 (due: TBA)</span></p><p class="c2"><span class="c1"></span></p><p class="c5"><span class="c9 c19">Quiz 8: Week 14 and 15 Lecture materials</span></p><p class="c2"><span class="c7"></span></p><p class="c2"><span class="c7"></span></p></body></html>